### DISCUSSION TITLE
GNNExplainer for multi-instance explanations

### DISCUSSION THREAD

**Discussion by shubhiambast** [pyg-team/pytorch_geometric#9518] [Q&A]
Hi, I want to use GNNExplainer for node classification task but aiming for explanations for each class. How could I do that? 
"In the case of multi-instance explanations, GNNEXPLAINER provides an explanation that consistently explains a set of instances (e.g., nodes of a given class)."

**Comment by kano5266** [comment]
`GNNExplainer` is mainly an instance-level explainer. 
For multi-instance explanations, you’ll need to choose a prototypical node for the class, use `GNNExplainer` to explain that node, and then apply graph alignment for all nodes in the class. 
These steps aren’t part of  the `GNNExplainer`, so you’ll need to write your own code to implement them. 
Check out Appendix A of the paper for more details.
